cmd/create/service/cmd.go

```diff
@@ -0,0 +1,360 @@
+	reporter := rprtr.CreateReporterOrExit()
+	logger := logging.CreateLoggerOrExit(reporter)
+
+	// Parse out CLI flags, then override positional arguments
+	// This allows for arbitrary flags used for addon parameters
+	_ = cmd.Flags().Parse(argv)
+
+	// Create the client for the OCM API:
+	ocmClient, err := ocm.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create OCM connection: %v", err)
+		os.Exit(1)
+	}
+	defer func() {
+		err = ocmClient.Close()
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	awsClient := aws.GetAWSClientForUserRegion(reporter, logger)
+
+	// Get AWS region
+	args.AwsRegion, err = aws.GetRegion(arguments.GetRegion())
+	if err != nil {
+		reporter.Errorf("Error getting region: %v", err)
+		os.Exit(1)
+	}
+	reporter.Debugf("Using AWS region: %s", args.AwsRegion)
+
+	awsCreator, err := awsClient.GetCreator()
+	if err != nil {
+		reporter.Errorf("Unable to get IAM credentials: %v", err)
+		os.Exit(1)
+	}
+	reporter.Debugf("Using AWS creator: %s", awsCreator.ARN)
+
+	args.AwsAccountID = awsCreator.AccountID
+	args.Properties = map[string]string{
+		properties.CreatorARN: awsCreator.ARN,
+		properties.CLIVersion: info.Version,
+	}
+
+	// Openshift version to use.
+	versionList, err := getVersionList(ocmClient)
+	if err != nil {
+		reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+	version := versionList[0]
+	minor := ocm.GetVersionMinor(version)
+
+	// Add-on parameter logic
+	addOn, err := ocmClient.GetAddOn(args.ServiceType)
+	if err != nil {
+		reporter.Errorf("Failed to get add-on '%s': %s", args.ServiceType, err)
+		os.Exit(1)
+	}
+	parameters := addOn.Parameters()
+
+	if parameters.Len() > 0 {
+		args.Parameters = map[string]string{}
+		// Determine if all required parameters have already been set as flags.
+		if arguments.HasUnknownFlags() {
+			parameters.Each(func(param *cmv1.AddOnParameter) bool {
+				flag := cmd.Flags().Lookup(param.ID())
+				if param.Required() && (flag == nil || flag.Value.String() == "") {
+					reporter.Errorf("Required parameter --%s missing", param.ID())
+					os.Exit(1)
+				}
+				if flag != nil {
+					val := strings.Trim(flag.Value.String(), " ")
+					if val != "" && param.Validation() != "" {
+						isValid, err := regexp.MatchString(param.Validation(), val)
+						if err != nil || !isValid {
+							reporter.Errorf("Expected %v to match /%s/", val, param.Validation())
+							os.Exit(1)
+						}
+					}
+					args.Parameters[param.ID()] = flag.Value.String()
+				}
+				return true
+			})
+		}
+	}
+
+	// Find all installer roles in the current account using AWS resource tags
+	var roleARN string
+	var supportRoleARN string
+	var controlPlaneRoleARN string
+	var workerRoleARN string
+
+	role := aws.AccountRoles[aws.InstallerAccountRole]
+
+	roleARNs, err := awsClient.FindRoleARNs(aws.InstallerAccountRole, minor)
+	if err != nil {
+		reporter.Errorf("Failed to find %s role: %s", role.Name, err)
```

Comment:
In this kind of messages i see that usually we enclose the subject between ticks, '%s', instead only %s. It seems that makes more easy to search in logs.
Apart of that, if we follow this pattern ("relevant element/subjects" tick enclosed), I  think we can use directly `%q` instead `'%s'` (this would be applied to all the messages in the MR)